kedro/framework/project/__init__.py

```diff
@@ -252,6 +252,11 @@ def configure_project(package_name: str):
     global PACKAGE_NAME
     PACKAGE_NAME = package_name
 
+    # Don't overwite any defined project log level in KEDRO_LOGGING_CONFIG
+    if PACKAGE_NAME and PACKAGE_NAME not in LOGGING["loggers"].keys():
+        LOGGING["loggers"].update({PACKAGE_NAME: {"level": "INFO"}})
+        LOGGING.configure(LOGGING.data)
+
```

Comment:
@antonymilne 
I guess this is the weirdest part in the code. I trace back to ParallelRunner, it passed the `LOGGING` as `logging_config` directly. So essentially it's doing something like `LOGGING.configure(LOGGING`. This isn't possible as it cause recursion error.

Reply:
I suspect the recursion error boils down to addressing because we are using the same object, but that's just a hunch. Using `LOGGING.data` successfully bypasses this.

Reply:
Can we do this as something like this instead?
```
# In ProjectLogging
    def set_project_logging(self, package_name):
        if package_name not in self.data["loggers"]:
            self.data["loggers"][package_name] = {"level": "INFO"}

# in configure_project
    LOGGING.set_project_logging(package_name)
```

Reply:
P.S. I know the `configure_logging` function is super weird but I don't think there's any other way to do it unfortunately. The parallel runner thing is a huge pain. I recommend not touching `configure_logging` at all if possible 😅